include/jemalloc/internal/mutex_pool_structs.h

```diff
@@ -0,0 +1,14 @@
+#ifndef JEMALLOC_INTERNAL_MUTEX_POOL_STRUCTS_H
+#define JEMALLOC_INTERNAL_MUTEX_POOL_STRUCTS_H
+
+/* This file really combines "structs" and "types", but only transitionally. */
+
+/* We do mod reductions by this value, so it should be kept a power of 2. */
+#define MUTEX_POOL_SIZE 256
```

Comment:
If handling allocation is a pain, we can consider defining a large array and just initialize the first N of them depending on ncpus (assuming we don't use huge page for them).